GUI support for decoupling sync privileges

raghavi101 opened the pull request.

anandf committed:
Initial attempt to support sa based sync

Signed-off-by: Anand Francis Joseph <anjoseph@redhat.com>

anandf committed:
Impersonation alternative implementation using destination serviceAccountName

Signed-off-by: Anand Francis Joseph <anjoseph@redhat.com>

ishitasequeira committed:
feat:Add cli sync with impersonate (#1)

* Add ServiceName to proj cli commands

* revert manifest changes

* revert manifest changes

raghavi101 committed:
GUI support for decoupling sync privileges

keithchong reviewed (commented):
Looks like it is a good start. You're prepping the UI to accept this new field for the service account name.  I see that Anand's repo has the new field in the Application Destination API object.

What exactly is the behaviour that you intend?

Eg. will you be able populate the list of service account names, and the user is able to choose one from the list?  There is also form validation that needs to be considered.

dependabot[bot] committed:
chore(deps): bump library/golang from 1.21.3 to 1.22.0 in /test/remote (#17111)

Bumps library/golang from 1.21.3 to 1.22.0.

---
updated-dependencies:
- dependency-name: library/golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump yarn from 1.22.10 to 1.22.13 in /ui-test (#17092)

Bumps [yarn](https://github.com/yarnpkg/yarn) from 1.22.10 to 1.22.13.
- [Release notes](https://github.com/yarnpkg/yarn/releases)
- [Changelog](https://github.com/yarnpkg/yarn/blob/master/CHANGELOG.md)
- [Commits](https://github.com/yarnpkg/yarn/compare/1.22.10...v1.22.13)

---
updated-dependencies:
- dependency-name: yarn
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: pasha-codefresh <pavel@codefresh.io>

dependabot[bot] committed:
chore(deps): bump library/redis from 7.0.11 to 7.2.4 in /test/container (#16806)

Bumps library/redis from 7.0.11 to 7.2.4.

---
updated-dependencies:
- dependency-name: library/redis
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: pasha-codefresh <pavel@codefresh.io>

dependabot[bot] committed:
chore(deps-dev): bump yarn from 1.22.10 to 1.22.21 in /ui (#17096)

Bumps [yarn](https://github.com/yarnpkg/yarn) from 1.22.10 to 1.22.21.
- [Release notes](https://github.com/yarnpkg/yarn/releases)
- [Changelog](https://github.com/yarnpkg/yarn/blob/master/CHANGELOG.md)
- [Commits](https://github.com/yarnpkg/yarn/compare/1.22.10...v1.22.21)

---
updated-dependencies:
- dependency-name: yarn
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Blake Pettersson <blake.pettersson@gmail.com>
Co-authored-by: pasha-codefresh <pavel@codefresh.io>

soravkumarsharma committed:
fix the typo (#17116)

dependabot[bot] committed:
chore(deps): bump library/node from 20.6.1 to 21.6.1 (#17053)

Bumps library/node from 20.6.1 to 21.6.1.

---
updated-dependencies:
- dependency-name: library/node
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: pasha-codefresh <pavel@codefresh.io>

fengshunli committed:
chore(deps): bump library/node from 20.7.0 to 21.6.1 (#17065)

Signed-off-by: fengshunli <1171313930@qq.com>
Co-authored-by: pasha-codefresh <pavel@codefresh.io>

raghavi101 committed:
Added UI support for new field 'destinationServiceAccounts' in AppProject.Spec

Signed-off-by: Raghavi Shirur <rshirur@redhat.com>

raghavi101 committed:
undo manifest error

Signed-off-by: Raghavi Shirur <rshirur@redhat.com>

prune998 committed:
fix(kustomize): set build dir (#15057) #16229 #16652 (#16653)

* use repo root, not app path

Signed-off-by: Prune <prune@lecentre.net>

correct patch

Signed-off-by: Prune <prune@lecentre.net>

* use Getwd to find the root path for diff commands

Signed-off-by: Prune <prune@lecentre.net>

* set dot a default for argo app commands

Signed-off-by: Prune <prune@lecentre.net>

* revert default values

Signed-off-by: Prune <prune@lecentre.net>

* patch diff in TestNamespacedResourceDiffing

Signed-off-by: Prune <prune@lecentre.net>

* patching some diff and sync

Signed-off-by: Prune <prune@lecentre.net>

* patch remaining diff in error

Signed-off-by: Prune <prune@lecentre.net>

---------

Signed-off-by: Prune <prune@lecentre.net>

blakepettersson committed:
chore(ci): tweak backend filters (#17134)

The existing backend filters get triggered even on frontend-only or
docs-only changes, which should not be the case. The reason for this
seems to be the fact that each filter line is ORed rather than ANDed.
To remedy this, we put all the filters on the same line.

I tried the filter out in a REPL
(https://runkit.com/blakepettersson/65c3daba99653f0008c74eda). This is
a filter using picomatch (the same library `dorny/paths-filter` uses).

Signed-off-by: Blake Pettersson <blake.pettersson@gmail.com>

agaudreault committed:
docs: fix error in toolchain setup (#17154)

Signed-off-by: Alexandre Gaudreault <alexandre_gaudreault@intuit.com>

ashutosh16 committed:
chore(dex): 2.37.0 to 2.38.0 (#17157)

Signed-off-by: asingh51 <ashutosh_singh@intuit.com>
Co-authored-by: asingh51 <ashutosh_singh@intuit.com>

shlomitubul committed:
feat: Add support for passing Redis Sentinel username(ACL) and password (#17168)

* Add support for passing Sentinel username and password

Signed-off-by: ShlomiTubul <shlomi.tubul@placer.ai>

* fix align with var naming

Signed-off-by: ShlomiTubul <shlomi.tubul@placer.ai>

* fix align with var naming

Signed-off-by: ShlomiTubul <shlomi.tubul@placer.ai>

---------

Signed-off-by: ShlomiTubul <shlomi.tubul@placer.ai>
Co-authored-by: ShlomiTubul <shlomi.tubul@placer.ai>

gdsoumya committed:
fix: stop initializing deployment informer if dynamic sharding is disabled (#17097)

* fix: stop initializing deployment informer if dynamic sharding is disabled

Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

* feat: updated sharding cache getter func

Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

---------

Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>
